x/evmutil/client/cli/evm.go

```diff
@@ -27,7 +27,12 @@ func CanSignEthTx(ctx client.Context) error {
 		return err
 	}
 
-	if keyInfo.GetAlgo() != hd.EthSecp256k1Type {
+	pubKey, err := keyInfo.GetPubKey()
+	if err != nil {
+		return err
+	}
+
+	if pubKey.Type() != string(hd.EthSecp256k1Type) {
 		return fmt.Errorf("from address does not support %v", hd.EthSecp256k1Type)
```

Comment:
ah I read it in the different way that seemed correct, "from address" referring to the input address instead of the requirement address type. But I can see how it is understood the other way, updated to be more clear without the ambiguity 😃